x-pack/plugins/observability/public/pages/alerts/alerts.test.tsx

```diff
@@ -0,0 +1,162 @@
+        observability: { enabled: false },
+      },
+      thresholdRule: { enabled: false },
+    },
+    compositeSlo: {
+      enabled: false,
+    },
+    aiAssistant: {
+      enabled: false,
+      feedback: {
+        enabled: false,
+      },
+    },
+  },
+  observabilityRuleTypeRegistry: createObservabilityRuleTypeRegistryMock(),
+  ObservabilityPageTemplate: KibanaPageTemplate,
+  kibanaFeatures: [],
+  core: {} as CoreStart,
+  plugins: {} as ObservabilityPublicPluginsStart,
+  hasAnyData: true,
+  isAllRequestsComplete: true,
+}));
+
+jest.spyOn(dataContext, 'useHasData').mockImplementation(() => ({
+  hasDataMap: {},
+  hasAnyData: true,
+  isAllRequestsComplete: true,
+  onRefreshTimeRange: jest.fn(),
+  forceUpdate: 'false',
+}));
+
+jest.mock('@kbn/alerts-ui-shared/src/maintenance_window_callout/api', () => ({
+  fetchActiveMaintenanceWindows: jest.fn(() => Promise.resolve([])),
+}));
+const fetchActiveMaintenanceWindowsMock = fetchActiveMaintenanceWindows as jest.Mock;
+
+jest.mock('../../hooks/use_time_buckets', () => ({
+  useTimeBuckets: jest.fn(),
+}));
+const { useTimeBuckets } = jest.requireMock('../../hooks/use_time_buckets');
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      retry: false,
+      cacheTime: 0,
+    },
+  },
+});
+function AllTheProviders({ children }: { children: any }) {
+  return (
+    <ThemeProvider
+      theme={() => ({ eui: { ...euiDarkVars, euiColorLightShade: '#ece' }, darkMode: true })}
+    >
+      <IntlProvider locale="en">
+        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+      </IntlProvider>
+    </ThemeProvider>
+  );
+}
+
+describe('AlertsPage with all capabilities', () => {
+  const timeBuckets = new TimeBuckets({
+    'histogram:maxBars': 12,
+    'histogram:barTarget': 10,
+    dateFormat: 'MMM D, YYYY @ HH:mm:ss.SSS',
+    'dateFormat:scaled': [
+      ['', 'HH:mm:ss.SSS'],
+      ['PT1S', 'HH:mm:ss'],
+      ['PT1M', 'HH:mm'],
+      ['PT1H', 'YYYY-MM-DD HH:mm'],
+      ['P1DT', 'YYYY-MM-DD'],
+      ['P1YT', 'YYYY'],
+    ],
+  });
+
+  async function setup() {
+    return render(<AlertsPage />, { wrapper: AllTheProviders });
+  }
+
+  beforeAll(() => {
+    fetchActiveMaintenanceWindowsMock.mockResolvedValue([]);
+  });
+
+  beforeEach(() => {
+    fetchActiveMaintenanceWindowsMock.mockClear();
+    useTimeBuckets.mockReturnValue(timeBuckets);
+  });
+
+  it('should render an alerts page template', async () => {
+    await act(async () => {
+      const wrapper = await setup();
+      expect(wrapper.queryByText('Alerts')).toBeInTheDocument();
+    });
+  });
+
+  it('renders MaintenanceWindowCallout if one exists', async () => {
+    fetchActiveMaintenanceWindowsMock.mockResolvedValue([RUNNING_MAINTENANCE_WINDOW_1]);
+
+    await act(async () => {
+      const wrapper = await setup();
+      expect(await wrapper.findByText('Maintenance window is running')).toBeInTheDocument();
```

Comment:
nit: Wouldn't it be better to use `data-test-subj` instead of finding an element with text?